library-app/src/components/pages/Login/Login.tsx

```diff
@@ -1,139 +1,118 @@
-    setEnteredEmail(currentTarget.value)
-    setEnteredEmailIsValid(true)
+  const passwordInputClasses = !enteredPasswordIsValid
+    ? `${styles['form-group']} ${styles['invalid']}`
+    : styles['form-group']
+
+  const navigateTo = useNavigate()
+
+  const getIsFieldValid = (term: string) => {
+    return term.trim() !== ''
   }
 
-  const handleOnBlurEmail = () => {
-    if (enteredEmail.trim() === '') {
-      setEnteredEmailIsValid(false)
-      return
-    }
+  const getIsFormValid = () => {
+    return getIsFieldValid(enteredEmail) && getIsFieldValid(enteredPassword)
   }
-  const handleOnBlurPassword = () => {
-    if (enteredPassword.trim() === '') {
-      setEnteredPasswordIsValid(false)
-      return
-    }
+
+  const handleEmailChange = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    setEnteredEmail(currentTarget.value.trim())
   }
 
   const handlePasswordChange = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
-    if (currentTarget.value.trim() === '') {
-      setEnteredPasswordIsValid(false)
-    }
-    setEnteredPassword(currentTarget.value)
-    setEnteredPasswordIsValid(true)
+    setEnteredPassword(currentTarget.value.trim())
   }
 
   const formSubmissionHandler = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault()
+    if (!getIsFormValid()) {
+      setEnteredEmailIsValid(getIsFieldValid(enteredEmail))
+      setEnteredPasswordIsValid(getIsFieldValid(enteredPassword))
+      return
+    }
     const formData: LoginRequestData = {
       email: enteredEmail,
       password: enteredPassword,
     }
-    if (!enteredEmailIsValid || !enteredPasswordIsValid) {
-      console.log('greska')
-    } else {
-      await LoginRequest(formData)
-        .then(({ data }) => {
-          localStorage.setItem('accessToken', data.accessToken)
-          localStorage.setItem('refreshToken', data.refreshToken)
-          localStorage.setItem('expiration', data.expiration)
-          navigateTo('/')
-          console.log(data)
-        })
-        .catch((error) => {
-          if (axios.isAxiosError(error)) {
-            if (error.response?.status === 401) {
-              setInvalidCredentials(true)
-              console.log('pogresni kredencijali')
-            }
-          }
-        })
-    }
-  }
 
-  const unmatchedCredentials = invalidCredentials
-  const unmatchedCredentialsClasses = unmatchedCredentials ? styles['invalid-credentials'] : ''
-
-  const emailInputIsInvalid = !enteredEmailIsValid
-  const emailInputClasses = emailInputIsInvalid
-    ? `${styles['form-group']} ${styles['invalid']}`
-    : styles['form-group']
-
-  const passwordInputIsInvalid = !enteredPasswordIsValid
-  const passwordInputClasses = passwordInputIsInvalid
-    ? `${styles['form-group']} ${styles['invalid']}`
-    : styles['form-group']
+    await LoginRequest(formData)
+      .then(({ data }) => {
+        setLocalStorage(data)
+        setIsLoggedIn(true)
+        setAccessToken(data.AccessToken)
```

Comment:
Do you set accessToken at setLocalStorage? If so, do you need this setAccessToken?

Reply:
I need it actually, because one is my state setter which i inherited through props, and other one is my from manageLocalStorage component.